examples/plugins/src/lighthouse/src/lighthouse.plugin.integration.test.ts

```diff
@@ -0,0 +1,144 @@
+import { lhr } from '../mock/fixtures/lhr';
+import { LIGHTHOUSE_OUTPUT_FILE_DEFAULT, corePerfGroupRefs } from './constants';
+import { audits, PLUGIN_SLUG as slug } from './index';
+import { create } from './lighthouse.plugin';
+
+describe('lighthouse-create-export', () => {
+  beforeEach(() => {
+    vol.fromJSON(
+      {
+        [LIGHTHOUSE_OUTPUT_FILE_DEFAULT]: JSON.stringify(lhr),
+      },
+      MEMFS_VOLUME,
+    );
+  });
+
+  it('should return valid PluginConfig if create is called', async () => {
+    const pluginConfig = await create({ url: LIGHTHOUSE_URL });
+    expect(() => pluginConfigSchema.parse(pluginConfig)).not.toThrow();
+    expect(pluginConfig).toEqual({
+      slug,
+      title: 'Lighthouse',
+      description: 'Chrome lighthouse CLI as code-pushup plugin',
+      icon: 'lighthouse',
+      runner: expect.any(Object),
+      audits,
+      groups: expect.any(Array),
+    });
+  });
+
+  it('should parse options for defaults correctly', async () => {
+    const pluginConfig = await create({
+      url: 'https://code-pushup.com',
+    });
+    expect(pluginConfig.runner.args).toEqual([
+      'lighthouse',
+      'https://code-pushup.com',
+      '--no-verbose',
+      '--output="json"',
+      '--output-path="lighthouse-report.json"',
+      '--onlyAudits="first-contentful-paint"',
+      '--onlyAudits="largest-contentful-paint"',
+      '--onlyAudits="speed-index"',
+      '--onlyAudits="total-blocking-time"',
+      '--onlyAudits="cumulative-layout-shift"',
+      '--onlyAudits="server-response-time"',
+      '--onlyAudits="interactive"',
+      '--chrome-flags="--headless=new"',
+    ]);
+  });
+
+  it('should parse options for headless by default to "new"', async () => {
+    const pluginConfig = await create({
+      url: LIGHTHOUSE_URL,
+    });
+    expect(pluginConfig.runner.args).toEqual(
+      expect.arrayContaining(['--chrome-flags="--headless=new"']),
+    );
+  });
+
+  it('should parse options for headless to new if true is given', async () => {
+    const pluginConfig = await create({
+      url: LIGHTHOUSE_URL,
+      headless: true,
+    });
+    expect(pluginConfig.runner.args).toEqual(
+      expect.arrayContaining(['--chrome-flags="--headless=new"']),
+    );
+  });
+
+  it('should parse options for headless to new if false is given', async () => {
+    const pluginConfig = await create({
+      url: LIGHTHOUSE_URL,
+      headless: false,
+    });
+    expect(pluginConfig.runner.args).toEqual(
+      expect.not.arrayContaining(['--chrome-flags="--headless=new"']),
+    );
+  });
+
+  it('should parse options for userDataDir correctly', async () => {
+    const pluginConfig = await create({
+      url: LIGHTHOUSE_URL,
+      userDataDir: 'test',
+    });
+    expect(pluginConfig.runner.args).toEqual(
+      expect.arrayContaining([
+        '--chrome-flags="--headless=new --user-data-dir=test"',
+      ]),
+    );
+  });
+
+  it('should return PluginConfig that executes correctly', async () => {
```

Comment:
Tests of such a different scope (parsing arguments vs executing the whole plugin) should not be in the same file under the same `describe`.

Reply:
done